### DISCUSSION TITLE
Repeated sprite colors

### DISCUSSION THREAD

**Discussion by spacek531** [OpenRCT2/OpenRCT2#16611] [Q&A]
https://github.com/OpenRCT2/OpenRCT2/blob/1062c3d08dde45fbd0a947921a6b0a94bcdf704e/src/openrct2/drawing/ImageImporter.h#L363-L375

Six of these ten colors are identical. This seems like a bad idea to me. Are these colors how they show up in-game? Seems like a good idea to differentiate the colors if it's possible.

**Comment by duncanspumpkin** [comment]
I suspect that the palette is based off of the first frame of the default palette.

**Comment by spacek531** [comment]
Every tool handles this differently:
- OpenRCT2 uses an AABCA pattern for both waves and sparkles
- Object Editor uses an AABCB pattern for both waves and sparkles
- Buggy's Ridemaker uses a ?AAAB pattern for both waves and sparkles
- RCT2RideGenerator uses the standard water colors but in reverse order
- landmark editor I didn't pay much attention to but it has a number of collisions

Aside from the collisions in each sequence:
- OpenRCT2's waves 0 collides with water tone 2
- Object Editor's waves 0 collides with water tone 2
- all of RCT2RideGenerator's waves and sparkles collide with the standard water colors

OpenRCT2 has modified base palette colors in the past (indices 1-9), I think this is another case where this should occur. When this gets the go-ahead, I will think about what colors should be replaced with.

**Comment by spacek531** [comment]
![image](https://user-images.githubusercontent.com/3454156/162128762-b029aae7-46b0-4b1a-bb92-166f009f654b.png)

I've made a modification (downstream of the second palette branch) that makes all base palette colors unique. I took the colors from watrcyan in no particular pattern, but I made the colors go in a wave pattern of brightness.
